python/pyspark/pandas/data_type_ops/num_ops.py

```diff
@@ -213,37 +215,43 @@ def abs(self, operand: IndexOpsLike) -> IndexOpsLike:
             F.abs(operand.spark.column), field=operand._internal.data_fields[0]
         )
 
+    def eq(self, left: IndexOpsLike, right: Any) -> SeriesOrIndex:
+        # We can directly use `super().eq` when given object is list, tuple, dict or set.
+        if not isinstance(right, IndexOpsMixin) and is_list_like(right):
+            return super().eq(left, right)
+        if _is_extension_dtypes(left) or _is_extension_dtypes(right):
+            return pyspark_column_op("__eq__", left, right)
+        return pyspark_column_op("__eq__", left, right, fillna=False)
+
+    def ne(self, left: IndexOpsLike, right: Any) -> SeriesOrIndex:
+        _sanitize_list_like(right)
+        if _is_extension_dtypes(left) or _is_extension_dtypes(right):
+            return pyspark_column_op("__ne__", left, right)
+        return pyspark_column_op("__ne__", left, right, fillna=True)
+
     def lt(self, left: IndexOpsLike, right: Any) -> SeriesOrIndex:
         _sanitize_list_like(right)
-        result = pyspark_column_op("__lt__")(left, right)
-        if is_remote():
-            # TODO(SPARK-43877): Fix behavior difference for compare binary functions.
-            result = result.fillna(False)
-        return result
+        if _is_extension_dtypes(left) or _is_extension_dtypes(right):
+            return pyspark_column_op("__lt__", left, right)
+        return pyspark_column_op("__lt__", left, right, fillna=False)
```

Comment:
I think you can put `_is_extension_dtypes(left) or _is_extension_dtypes(right)` condition to `fillna`?

Reply:
Sounds good. Just moved the logic into util.